sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,681 @@
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    n_components = rand_data.n_components
+    n_init = 10
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        g = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type)
+        ll = []
+        for _ in range(n_init):
+            g.fit(X)
+            ll.append(g.score(X))
+        ll = np.array(ll)
+        g_best = GaussianMixture(n_components=n_components,
+                                 n_init=n_init, max_iter=100, reg_covar=0,
+                                 random_state=rng, covariance_type=cov_type)
+        g_best.fit(X)
+        assert_almost_equal(ll.min(), g_best.score(X))
+
+
+def test_gaussian_mixture_fit_convergence_warning():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    n_components = rand_data.n_components
+    max_iter = 1
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        g = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=max_iter, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type)
+        assert_warns_message(ConvergenceWarning,
+                             'Initialization %d is not converged. '
+                             'Try different init parameters, '
+                             'or increase n_init, tol '
+                             'or check for degenerate data.'
+                             % max_iter, g.fit, X)
+
+
+def test_gaussian_mixture_n_parameters():
+    # Test that the right number of parameters is estimated
+    rng = np.random.RandomState(0)
+    n_features, n_components = 5, 2
+    n_params = {'spherical': 13, 'diag': 21, 'tied': 26, 'full': 41}
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.n_features = n_features
+        assert_equal(g._n_parameters(), n_params[cv_type])
+
+
+def test_gaussian_mixture_aic_bic():
+    # Test the aic and bic criteria
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 50, 3, 2
+    X = rng.randn(n_samples, n_features)
+    # standard gaussian entropy
+    SGH = 0.5 * (np.log(np.linalg.det(np.cov(X.T, bias=1))) +
+                 n_features * (1 + np.log(2 * np.pi)))
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.fit(X)
+        aic = 2 * n_samples * SGH + 2 * g._n_parameters()
+        bic = (2 * n_samples * SGH +
+               np.log(n_samples) * g._n_parameters())
+        bound = n_features * 3. / np.sqrt(n_samples)
+        assert_true(np.abs(g.aic(X) - aic) / n_samples < bound)
```

Comment:
This bound is also defined in the old `test_gmm.py`, I thought it was a know property of BIC and AIC.
@bthirion Do you know where `bound` comes from ?